dockerfiles/hsm-support/Dockerfile

```diff
@@ -2,14 +2,27 @@ FROM buildenv
 
 # Install MoCOCrW dependencies (except OpenSSL)
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y --no-install-recommends install \
-        # libp11 engine
-        libengine-pkcs11-openssl \
-        # headers for p11 engine
-        libp11-dev \
         # for pkcs11-tool which we use to create keys in token
         opensc \
         # p11-kit-modules allows loading of libp11 engine without having to edit openssl.cnf
         p11-kit-modules \
         # softhsm2: includes both softhsm2-util and libsofthsm2
         softhsm2 \
+        # libp11 needs this
+        libtool \
         && rm -rf /var/lib/apt/lists/*
+
+ADD keygen.patch /tmp/
+
+RUN cd /tmp && \
+    wget https://github.com/OpenSC/libp11/releases/download/libp11-0.4.12/libp11-0.4.12.tar.gz && \
+    tar xvf libp11-0.4.12.tar.gz && \
+    cd libp11-0.4.12 && \
+    git apply /tmp/keygen.patch && \
+    echo "Successfully patched libp11" && \
+    autoreconf --verbose --install --force && \
+    ./configure --enable-strict && \
+    make -j$(nproc) && \
+    make check && \
+    make install
```

Comment:
you probably want to do all of this in a subdir of /tmp that can later be removed again to reduce the size of the resulting container.

Reply:
Done